### PR TITLE
Add CI to next branch.

### PR DESCRIPTION
**Pull request by sagudev**
Suppressing #36 

TODO:
- Later when we catch ESR we can move next branch to follow `mozilla-release`. The we could also lower cron.
- Replace version in mason build with mozjs build?

### CODE REVIEW

**Comment by sagudev** [comment]
Should fix #12

File: tools/generic_lib.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+sed -i 's/mozjs-$MOZILLA_SYMBOLVERSION/mozjs/g' old-configure.in
```
Comment (ptomato) [inline tools/generic_lib.sh]: Oh, this is a nice trick.

**Comment by ptomato** [comment]
What does the workflow look like when we make `next` into the `esr91` branch? Do we branch `esr91` off of `next` and then add a commit on `esr91` removing this?

**Comment by sagudev** [comment]
> What does the workflow look like when we make `next` into the `esr91` branch? Do we branch `esr91` off of `next` and then add a commit on `esr91` removing this?

Yes, that was my plan. Maybe this should be documented. There is also change needed in https://github.com/sagudev/spidermonkey-embedding-examples/blob/patch-1/tools/get_sm.sh#L3

**Comment by sagudev** [comment]
On second thought, maybe setting env var in workflow would be better for `get_sm` script, but then esr branches contain wrong script, so never-mind.

**Comment by moztcampbell** [comment]
One thing I wonder about with the sed trick is that local builds will end up with mismatched version numbers.

**Comment by sagudev** [comment]
> One thing I wonder about with the sed trick is that local builds will end up with mismatched version numbers.

This sounds like we need another sed trick for meson.build 😄

**Comment by moztcampbell** [comment]
> 
> 
> > One thing I wonder about with the sed trick is that local builds will end up with mismatched version numbers.
> 
> This sounds like we need another sed trick for meson.build 😄

yeah, that is what I wonder. Locally builds would occasionally need to bump the version, but at least we wouldn't have CI always be red

**Comment by moztcampbell** [comment]
Let's switch back to beta and then we can give this a try.

**Comment by sagudev** [comment]
Ready for merge!

**Review by moztcampbell — APPROVED** [review]
Excellent! thanks for figuring this out

**Pull request merged**